smartsim/_core/arguments/shell.py

```diff
@@ -0,0 +1,48 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+
+import typing as t
+from abc import abstractmethod
+
+from smartsim.log import get_logger
+from smartsim.settings.arguments.launchArguments import LaunchArguments
+
+if t.TYPE_CHECKING:
+    from smartsim.settings.arguments import LaunchArguments
+
+logger = get_logger(__name__)
+
+
+class ShellLaunchArguments(LaunchArguments):
+    @abstractmethod
+    def format_env_vars(
+        self, env_vars: t.Dict[str, t.Optional[str]]
+    ) -> t.Union[t.List[str], None]: ...
+
+    @abstractmethod
+    def format_launch_args(self) -> t.Union[t.List[str], None]: ...
```

Comment:
Very minor nit: while we're here can we upgrade to the new style type hints?
```suggestion
    @abstractmethod
    def format_env_vars(
        self, env_vars: dict[str, str | None]
    ) -> list[str] | None: ...

    @abstractmethod
    def format_launch_args(self) -> list[str] | None: ...
```